packages/edit-post/src/editor.js

```diff
@@ -191,10 +175,26 @@ function Editor( { postId, postType, settings, initialEdits, ...props } ) {
 			} );
 		}
 
-		return hasThemeStyles && themeStyles.length
-			? settings.styles
-			: defaultEditorStyles;
-	}, [ settings, hasThemeStyles ] );
+		result.styles =
+			hasThemeStyles && themeStyles.length
```

Comment:
`hasThemeStyles` flag corresponds to a user preferences that is changeable from the preferences modal. That means that any change to the "settings" that is done by `updateEditorSettings` call from plugin authors will override any change that this preference might have.

Meaning we might want to avoid doing anything here to the "styles" property and instead move the logic of this preference to within the `EditorProvider` component (potentially need to make hasThemeStyles a new setting in editorSettings though).

Maybe there are other options, but it does raise the question about whether "styles" was ever meant to be "editable".